operator/pkg/controllers/controller.go

```diff
@@ -48,40 +48,43 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 		if errors.IsNotFound(err) {
 			return reconcile.Result{}, nil
 		}
-		return *status.Failed, err
+		return *result.Failed, err
 	}
 	// 2. Copy object for merge patch base
 	persisted := resource.DeepCopyObject()
 	// 3. Reconcile else finalize if object is deleted
-	result, reconcileErr := c.reconcile(ctx, resource, persisted)
+	reconcileResult, reconcileErr := c.reconcile(ctx, resource, persisted)
 	// 4. Update Status using a merge patch, we want to set status even when reconcile errored
 	if err := c.Status().Patch(ctx, resource, client.MergeFrom(persisted)); err != nil && !errors.IsNotFound(err) {
-		return *status.Failed, fmt.Errorf("status patch for %s, %w,", req.NamespacedName, err)
+		return *result.Failed, fmt.Errorf("status patch for %s, %w,", req.NamespacedName, err)
 	}
 	if reconcileErr != nil {
-		return *status.Failed, reconcileErr
+		if errors.IsWaitingForSubResource(reconcileErr) {
+			return *result.Waiting, nil
+		}
+		return *result.Failed, reconcileErr
 	}
-	return result, nil
+	return reconcileResult, nil
 }
 
 func (c *GenericController) reconcile(ctx context.Context, resource Object, persisted runtime.Object) (reconcile.Result, error) {
-	var result *reconcile.Result
+	var reconcileResult *reconcile.Result
```

Comment:
See earlier comment - I think having to do this renaming is probably a sign that that `result` pkg's name should change instead.